flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/logical/utils/LogicalTypeChecks.java

```diff
@@ -145,6 +156,14 @@ public static boolean isSingleFieldInterval(LogicalType logicalType) {
 		return logicalType.accept(SINGLE_FIELD_INTERVAL_EXTRACTOR);
 	}
 
+	public static boolean areTypeEqualsWithoutNames(LogicalType thisType, LogicalType thatType) {
```

Comment:
nit: rename to `hasEqualTypesWithoutNames`

Reply:
What's the reason to use `hasEqualTypesWithoutNames`? It looks like whether it contains some equal types (might children types are equal).